### ISSUE TITLE
Is proxywrap working properly with https module?

### ISSUE THREAD

**Issue by diegossilveira** [findhit/proxywrap#25]
Hello,

I'm testing proxywrap module, which worked well with node's http module. When I moved to https module, it doesn't get the correct remote address from proxy-protocol.

See this code snippet (which runs behind an HA proxy with proxy-protocol enabled):

``` javascript
const https = require('https')
const proxiedHttp = require('findhit-proxywrap').proxy(https)

const options = {
  key: <some key file>
  cert: <some cert file>
}

proxiedHttp.createServer(options, (req, res) => {
  console.log(req.connection.remoteAddress) // prints the HA proxy IP address
  res.end()
}).listen(8443)
```

Note: I'm using node 6.5.0

**Comment by cusspvz** [comment]
Unfortunately there is an issue with the http2 module yet. Because `http2` creates a new socket and places http underneath you have to access `remoteAddress` from `req.socket._parent`. We use tls on the HAProxy, so we aren't encountering this issue, but if you want to PR with a solution, I would appreciate it. :)

**Comment by diegossilveira** [comment]
Thanks for the reply! Me and a colleague just found a way to "solve" this issue and he will create a PR as soon as possible. The bad thing about this solution is that it relies on a "private"member of node's `tls.TLSSocket` (`_parent`), which wraps the `net.Socket` when `https.Server` is used, and we have no guarantee that it will not change in the future. But we made some tests and it works very well.

**Comment by cusspvz** [comment]
Nice @diegossilveira!! I will be waiting for the PR though I think there could be other ways to detect the `https.Server` instances with `instanceof`.

Gotta ask, is there any motif that impels you to use tls termination on the LB instead?
In all terms, having performance included, in case you have a LB on top of node the best TLS solution is on the LB. HTTPS on internal networks is a bottleneck (supposing your internal network will always be secured).

**Comment by diegossilveira** [comment]
PR #26 was created, hope this solves the issue.

BTW, we are using Amazon Elastic Load Balancer and, when using layer 7 balancing mode with SSL, it requires all upstream nodes are served with SSL too. So we opted to using a layer 4 balancing mode and have upstream nodes serving HTTPs.

**Comment by cusspvz** [comment]
released at `v0.3.12`
